clusters/resource_cluster.go

```diff
@@ -363,38 +439,54 @@ func resourceClusterUpdate(ctx context.Context, d *schema.ResourceData, c *commo
 		// don't add externally added libraries, if config has no `library {}` blocks
 		return nil
 	}
-	var libraryList libraries.ClusterLibraryList
+	var libraryList libraries.LibraryList
+	libraryList.ClusterId = clusterId
 	common.DataToStructPointer(d, clusterSchema, &libraryList)
-	librariesAPI := libraries.NewLibrariesAPI(ctx, c)
-	libsClusterStatus, err := librariesAPI.ClusterStatus(clusterID)
+	libsClusterStatus, err := w.Libraries.ClusterStatusByClusterId(ctx, clusterId)
+	if err != nil {
+		return err
+	}
+	libsToInstall, libsToUninstall := libraries.GetLibrariesToInstallAndUninstall(libraryList, libsClusterStatus)
+	clusterInfo, err = clusters.GetByClusterId(ctx, clusterId)
+	err = wrapMissingClusterError(err, d.Id())
 	if err != nil {
 		return err
 	}
```

Comment:
It looks very wrong to not have `if err != nil` immediately after it is returned.
```suggestion
	if err != nil {
		return wrapMissingClusterError(err, d.Id())
	}
```